packages/block-editor/src/components/block-tools/insertion-point.js

```diff
@@ -229,9 +230,25 @@ function InsertionPointPopover( {
 }
 
 export default function InsertionPoint( props ) {
-	const isVisible = useSelect( ( select ) => {
-		return select( blockEditorStore ).isBlockInsertionPointVisible();
+	const { insertionPoint, isVisible } = useSelect( ( select ) => {
+		const { getBlockInsertionPoint, isBlockInsertionPointVisible } =
+			select( blockEditorStore );
+		return {
+			insertionPoint: getBlockInsertionPoint(),
+			isVisible: isBlockInsertionPointVisible(),
+		};
 	}, [] );
 
-	return isVisible && <InsertionPointPopover { ...props } />;
+	return (
+		isVisible &&
+		( insertionPoint.operation === 'replace' ? (
```

Comment:
It could be worth adding a small comment here along the lines of "When replacing, use a popover that overlays the block, and when inserting use a popover that appears between blocks."